File: data/code_pairs/fortran/766.f90
! kinds_module.f90
module kinds_module
  implicit none
contains
  subroutine test_double()
    print *, "test_double called"
  end subroutine test_double

  subroutine test_complex()
    print *, "test_complex called"
  end subroutine test_complex
end module kinds_module

! bhmie_module.f90
module bhmie_module
  implicit none
contains
  subroutine bhmie_driver()
    print *, "bhmie_driver called"
  end subroutine bhmie_driver
end module bhmie_module

! dmiess_module.f90
module dmiess_module
  implicit none
contains
  subroutine dmiess_driver()
    print *, "dmiess_driver called"
  end subroutine dmiess_driver
end module dmiess_module

! dmilay_module.f90
module dmilay_module
  implicit none
contains
  subroutine dmilay_driver()
    print *, "dmilay_driver called"
  end subroutine dmilay_driver
end module dmilay_module

! test_modules.f90
program test_modules
  use kinds_module
  use bhmie_module
  use dmiess_module
  use dmilay_module
  implicit none

  call test_double()
  call test_complex()
  call bhmie_driver()
  call dmiess_driver()
  call dmilay_driver()

end program test_modules

File: data/code_pairs/cpp/766.cpp
// kinds_module.h
namespace kinds_module {
    void test_double() {
        std::cout << "test_double called" << std::endl;
    }
    void test_complex() {
        std::cout << "test_complex called" << std::endl;
    }
}

// bhmie_module.h
namespace bhmie_module {
    void bhmie_driver() {
        std::cout << "bhmie_driver called" << std::endl;
    }
}

// dmiess_module.h
namespace dmiess_module {
    void dmiess_driver() {
        std::cout << "dmiess_driver called" << std::endl;
    }
}

// dmilay_module.h
namespace dmilay_module {
    void dmilay_driver() {
        std::cout << "dmilay_driver called" << std::endl;
    }
}

// main.cpp (or an appropriate implementation file)
#include "kinds_module.h"
#include "bhmie_module.h"
#include "dmiess_module.h"
#include "dmilay_module.h"
#include <iostream>
#include <functional>

int main() {
    // Direct calls to simulate the behavior of the Fortran program
    kinds_module::test_double();
    kinds_module::test_complex();
    bhmie_module::bhmie_driver();
    dmiess_module::dmiess_driver();
    dmilay_module::dmilay_driver();

    return 0;
}